### PR TITLE
Cloud runner develop - latest fixes

### PR DESCRIPTION
**Pull request by frostebite**
#### Changes

- General cleanup, refinement and fixes.
- K8s stability significantly improved.
- Greatly improved pipeline with quick response smoke tests before long-running integration tests.
- Removed parameter options that didn't work well e.g. slow "constant garbage collection".
- Naming improvements for parameters.
- Cleanup environment variable injection, no longer duplicates into two formats.

#### Checklist

- [x] Read the contribution [guide](https://github.com/game-ci/unity-builder/blob/main/CONTRIBUTING.md) and accept the
      [code](https://github.com/game-ci/unity-builder/blob/main/CODE_OF_CONDUCT.md) of conduct
- [ ] Readme (updated or not needed)
- [ ] Tests (added, updated or not needed)

### CODE REVIEW

**Review by AndrewKahr — DISMISSED** [review]
Overall lgtm! Only thing that I noticed was a lot of type hints were removed which would be nice to keep as what I noticed when I had gone through the repo is it's very difficult to make changes to this very complex and dense code. Type hints go a long way towards helping understand what is going on in the code and help reduce the chances of logic errors as the compiler will verify usage.

**Comment by frostebite** [comment]
@AndrewKahr all resolved, thanks again!

**Comment by AndrewKahr** [comment]
@frostebite I still see several unresolved comments. Did you get a chance to look those over? You might need to uncollapse the conversation as github auto collapsed most of the comments. If you go to the files changed tab I believe it should be easier to see the comments throughout the files.

**Comment by frostebite** [comment]
> @frostebite I still see several unresolved comments. Did you get a chance to look those over? You might need to uncollapse the conversation as github auto collapsed most of the comments. If you go to the files changed tab I believe it should be easier to see the comments throughout the files.

Ah they were just hidden in the GitHub UI. I will resolve them all too.

File: src/model/build-parameters.ts [hunk 1]
```diff
@@ -70,15 +70,21 @@ class BuildParameters {
   public cloudRunnerDebug!: boolean | undefined;
   public cloudRunnerBuilderPlatform!: string | undefined;
   public isCliMode!: boolean;
-  public retainWorkspace!: boolean;
-  public maxRetainedWorkspaces!: number;
+  public retainWorkspaces!: number;
   public useSharedLargePackages!: boolean;
-  public useLz4Compression!: boolean;
+  public compressionStrategy!: boolean;
```
[the remark this answers is not in the document]
Comment (frostebite) [reply src/model/build-parameters.ts:74]: Ultimately yes I want to support several strategies. 

"Folder Sync" "Tar Archive" "Lz4 fast compression" "Zip smallest possible compression"

For now only "lz4" or the default "tar" is supported, so there is only one compression strategy to use?

File: src/model/cloud-runner/cloud-runner-options.ts [hunk 2]
```diff
@@ -225,27 +239,21 @@ class CloudRunnerOptions {
 
   static get cloudRunnerDebug(): boolean {
     return (
-      CloudRunnerOptions.getInput(`cloudRunnerTests`) === 'true' ||
-      CloudRunnerOptions.getInput(`cloudRunnerDebug`) === 'true' ||
+      CloudRunnerOptions.getInput(`cloudRunnerTests`) === `true` ||
+      CloudRunnerOptions.getInput(`cloudRunnerDebug`) === `true` ||
+      CloudRunnerOptions.getInput(`cloudRunnerDebug`) === `true` ||
```
[the remark this answers is not in the document]
Comment (frostebite) [reply src/model/cloud-runner/cloud-runner-options.ts:244]: Was meant to cover the replaced "Tree" input.

File: src/model/cloud-runner/cloud-runner-options.ts [hunk 3]
```diff
@@ -269,22 +277,14 @@ class CloudRunnerOptions {
   // Retained Workspace
   // ### ### ###
 
-  public static get retainWorkspaces(): boolean {
-    return CloudRunnerOptions.getInput(`retainWorkspaces`) === 'true' || false;
-  }
-
-  static get maxRetainedWorkspaces(): number {
-    return Number(CloudRunnerOptions.getInput(`maxRetainedWorkspaces`)) || 3;
+  public static get retainWorkspaces(): number {
```
[the remark this answers is not in the document]
Comment (frostebite) [reply src/model/cloud-runner/cloud-runner-options.ts:280]: renamed

File: src/model/build-parameters.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (frostebite) [reply src/model/build-parameters.ts:74]: Renamed to useCompressionStrategy for now

**Comment by frostebite** [comment]
@AndrewKahr Should all be resolved now, thanks! :)

**Review by webbertakken — APPROVED** [review]

**Review by AndrewKahr — APPROVED** [review]
LGTM, just the one comment on clearing specific cloud runner fields.

File: src/model/cloud-runner/cloud-runner.ts [hunk 4]
```diff
@@ -95,21 +110,21 @@ class CloudRunner {
...
-        CloudRunner.buildParameters.buildGuid,
-        CloudRunner.buildParameters,
-        CloudRunner.buildParameters.branch,
-        CloudRunner.defaultSecrets,
-      );
-      if (!CloudRunner.buildParameters.isCliMode) core.endGroup();
-      await GitHub.updateGitHubCheck(CloudRunner.buildParameters.buildGuid, CloudRunner.buildParameters.buildGuid);
+      const content = { ...CloudRunner.buildParameters };
+      content.gitPrivateToken = ``;
+      content.unitySerial = ``;
```
[the remark this answers is not in the document]
Comment (AndrewKahr) [reply src/model/cloud-runner/cloud-runner.ts:119]: Oh that's mb then, thought it was bringing through all the secrets. It should be good to go then.

**Pull request merged**